radio/src/gui/colorlcd/topbar.cpp

```diff
@@ -83,8 +82,5 @@ void SetupTopBarWidgetsPage::deleteLater(bool detach, bool trash)
   // and continue async deletion...
   FormWindow::deleteLater(detach, trash);
 
-  // restore screen setting tab on top
-  new ScreenMenu(0);
-
```

Comment:
This is the culprit - on closing the `SetupTopBarWidgetsPage` page the first screen setting page isn't being re-created... just like in `widgets_setup.cpp deleteLater`

What I don't know is if this was part of the memory leak you were chasing, or wrong place at the wrong time ;)

Reply:
This is right: this is basically what sends the user back to the "User Interface" page, so it needs to stay.